diff --git a/includes/behaviors/js/openlayers_behavior_geofield.js b/includes/behaviors/js/openlayers_behavior_geofield.js
index 9c053eb..72b5ff1 100644
--- a/includes/behaviors/js/openlayers_behavior_geofield.js
+++ b/includes/behaviors/js/openlayers_behavior_geofield.js
@@ -29,17 +29,28 @@ OpenLayers.Control.GeofieldEditingToolbar = OpenLayers.Class(
         var tool = null;
 
         if (options.allow_edit && options.allow_edit !== 0) {
+
+          var deleteFeatureControl = new OpenLayers.Control.SelectFeature(layer, {
+            clickout: false,
+            toggle: false,
+            title: "Delete",
+            displayClass: "olControlDelete"
+          });
+
+          deleteFeatureControl.events.register("featurehighlighted", this, function(e) {
+            if (confirm('Are you sure you want to delete this feature?')) {
+              layer.removeFeatures([e.feature]);
+              deleteFeatureControl.deactivate();
+            } else {
+              deleteFeatureControl.unselect(e.feature);
+            }
+          });
+
+          controls.push(deleteFeatureControl);
+
           // add an Edit feature
           controls.push(new OpenLayers.Control.ModifyFeature(layer, {
-            deleteCodes: [46, 68, 100],
-            handleKeypress: function(evt) {
-              if (this.feature && OpenLayers.Util.indexOf(this.deleteCodes, evt.keyCode) > -1) {
-                // We must unselect the feature before we delete it
-                var feature_to_delete = this.feature;
-                this.selectControl.unselectAll();
-                this.layer.removeFeatures([feature_to_delete]);
-              }
-            }
+            deleteCodes: [46, 68, 100]
           }));
         } else {
           controls = [new OpenLayers.Control.Navigation()];
@@ -84,7 +95,7 @@ OpenLayers.Control.GeofieldEditingToolbar = OpenLayers.Class(
    */
   Drupal.behaviors.openlayers_behavior_geofield = {
     'attach': function(context, settings) {
-      
+
       // Only attach the behavior to a map
       if (!$(context).hasClass('openlayers-map')) return;
 
