diff --git a/plugins/display_renderers/panels_renderer_editor.class.php b/plugins/display_renderers/panels_renderer_editor.class.php
index 4876ff3..0ea00b3 100644
--- a/plugins/display_renderers/panels_renderer_editor.class.php
+++ b/plugins/display_renderers/panels_renderer_editor.class.php
@@ -467,21 +467,24 @@ class panels_renderer_editor extends panels_renderer_standard {
     drupal_alter('get_pane_links', $links, $pane, $content_type);
 
     $dropdown_links = $links['top'];
-    foreach (array(
-  'style' => 'Style',
-  'visibility' => 'Visibility rules',
-  'lock' => 'Locking',
-  'cache' => 'Caching'
-    ) as $category => $label) {
-      $dropdown_links[] = array(
-        'title' => '<hr />',
-        'html' => TRUE,
-      );
-      $dropdown_links[] = array(
-        'title' => '<span class="dropdown-header">' . t($label) . '</span>' . theme_links(array('links' => $links[$category], 'attributes' => array(), 'heading' => array())),
-        'html' => TRUE,
-        'attributes' => array('class' => array('panels-sub-menu')),
-      );
+    $category_labels = array(
+      'style' => 'Style',
+      'visibility' => 'Visibility rules',
+      'lock' => 'Locking',
+      'cache' => 'Caching',
+    );
+    foreach ($category_labels as $category => $label) {
+      if (array_key_exists($category, $links)) {
+        $dropdown_links[] = array(
+          'title' => '<hr />',
+          'html' => TRUE,
+        );
+        $dropdown_links[] = array(
+          'title' => '<span class="dropdown-header">' . t($label) . '</span>' . theme_links(array('links' => $links[$category], 'attributes' => array(), 'heading' => array())),
+          'html' => TRUE,
+          'attributes' => array('class' => array('panels-sub-menu')),
+        );
+      }
     }
 
     $dropdown_links[] = array(
